src/Metadata/Property/Factory/ExtractorPropertyMetadataFactory.php

```diff
@@ -146,13 +146,15 @@ private function createSubresourceMetadata($subresource, PropertyMetadata $prope
         if (null !== $type) {
             $isCollection = $type->isCollection();
             $resourceClass = $isCollection ? $type->getCollectionValueType()->getClassName() : $type->getClassName();
+            $maxDepth = null; // for Type we can't configure maxDepth, maxDepth is always null
```

Comment:
can you declare this var l145 instead? 

we don't need the comment, null is default anyway